crates/re_types/src/datatypes/angle.rs

```diff
@@ -101,50 +101,26 @@ impl ::re_types_core::Loggable for Angle {
             let fields = vec![
                 NullArray::new(DataType::Null, data.iter().filter(|v| v.is_none()).count()).boxed(),
                 {
-                    let (somes, radians): (Vec<_>, Vec<_>) = data
+                    let radians: Vec<_> = data
                         .iter()
-                        .filter(|datum| matches!(datum.as_deref(), Some(Angle::Radians(_))))
-                        .map(|datum| {
-                            let datum = match datum.as_deref() {
-                                Some(Angle::Radians(v)) => Some(v.clone()),
-                                _ => None,
-                            };
-                            (datum.is_some(), datum)
+                        .filter_map(|datum| match datum.as_deref() {
+                            Some(Angle::Radians(v)) => Some(v.clone()),
+                            _ => None,
                         })
-                        .unzip();
-                    let radians_bitmap: Option<arrow2::bitmap::Bitmap> = {
-                        let any_nones = somes.iter().any(|some| !*some);
-                        any_nones.then(|| somes.into())
-                    };
-                    PrimitiveArray::new(
-                        DataType::Float32,
-                        radians.into_iter().map(|v| v.unwrap_or_default()).collect(),
-                        radians_bitmap,
-                    )
-                    .boxed()
+                        .collect();
+                    PrimitiveArray::new(DataType::Float32, radians.into_iter().collect(), None)
```

Comment:
We should be able to collect into an `arrow2::buffer::Buffer` directly, without the extra collect to a `Vec`, no?

Reply:
I think so too, but it's one of the many things I struggled to get rid of

Reply:
I can give it a shot. This will otherwise result in a performance regression

Reply:
I unfortunately didn't run any benchmarks :((, but I figured it wouldn't because before we did `radians.into_iter().map(|v| v.unwrap_or_default()).collect(),` where `radians` was already collected, similiar to how it is now.

In any case I would very much appreciate a cleanup attempt from your side, been struggling with this thing a lot. Let me know if you want to pair up!